File: Python/address/create_address.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.models import *
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)

customers_controller = customers_controller.CustomersController()

customerId = "cus_PzRyp10FeNca2rVB"

request = create_address_request.CreateAddressRequest()

request = create_address_request.CreateAddressRequest()
request.line_1 = "10880, Malibu Point, Malibu Central"
request.line_2 = "7º floor"
request.zip_code = "90265"
request.city = "Malibu"
request.state = "CA"
request.country = "US"
request.metadata = update_metadata_request.UpdateMetadataRequest()
request.metadata.id = "my_address_id"

try:
    result = customers_controller.create_address(customerId, request)
    assert result is not None
    assert result.id is not None
    print("Address id: ", result.id)

except ErrorException as ex:
    print(ex.message)
    print("Errors: ", ex.errors)
except Exception as ex:
    raise ex

File: Java/address/CreateAddress.java
package examples.address;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.CustomersController;
import me.pagar.api.http.client.APICallBack;
import me.pagar.api.http.client.HttpContext;
import me.pagar.api.models.*;

import java.util.LinkedHashMap;

public class CreateAddress {

    public static void main(String[] args) {

        String basicAuthUserName = "ak_test_4tdVXpseumRmqbo"; // The username to use with basic authentication
        String basicAuthPassword = ""; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        CustomersController customers_controller = new CustomersController();

        String customerId = "cus_PzRyp10FeNca2rVB";

        CreateAddressRequest request = new CreateAddressRequest();

        request.setLine1("10880, Malibu Point, Malibu Central");
        request.setLine2("7º floor");
        request.setZipCode("90265");
        request.setCity("Malibu");
        request.setState("CA");
        request.setCountry("US");
        LinkedHashMap<String, String> metadata = new LinkedHashMap<String, String>();
        metadata.put("id", "my_address_id");
        UpdateMetadataRequest updateMetadata = new UpdateMetadataRequest();
        updateMetadata.setMetadata(metadata);
        request.setMetadata(updateMetadata.getMetadata());

        customers_controller.createAddressAsync(customerId, request, null, new APICallBack<GetAddressResponse>() {
            @Override
            public void onSuccess(HttpContext context, GetAddressResponse response) {

                System.out.println("Address create !");
                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println("Address id: " + response.getId());

            }

            @Override
            public void onFailure(HttpContext context, Throwable error) {

                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println(error.getMessage());
                error.printStackTrace();

            }
        });
    }
}